### PR TITLE
Added NotificationIcon component

### LINKED ISSUE #608: Create a notification icon component with red badge
Simulator Screen Shot - iPhone 14 - 2023-08-31 at 15 11 31

### PR DESCRIPTION
**Pull request by mohitharshan123**
- Fixes #608 
@muhammedjasircp _A please review

**Checklist**

- [x] I have performed a self-review of my code and tested well before raising the PR.
- [x] I have made corresponding changes to the documentation and `index.d.ts`.
- [x] I've verified the change via .yalc in some neetoApp.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -66,11 +66,11 @@ const theme: {
...
-    SpaceProps,
-    BorderProps,
-    ColorProps,
-    LayoutProps,
-    PositionProps {
+  SpaceProps,
+  BorderProps,
+  ColorProps,
+  LayoutProps,
+  PositionProps {
```
Comment (muhammedjasircp) [inline index.d.ts:73]: There seems to be a prettier issue here

**Comment by sangameshsomawar** [comment]
@mohitharshan123 We don't need to push lib changes explicitly, the library has a [workflow script](https://github.com/bigbinary/neeto-ui-rn/blob/main/.github/workflows/publish.yml#L14)  which builds on merge to main branch.

**Comment by muhammedjasircp** [comment]
@sangameshsomawar won't it be better to add lib folder to gitignore

**Comment by sangameshsomawar** [comment]
@muhammedjasircp Previously we thought to keep lib changes in git. Now I feel we don't need lib in git. We can add in gitignore.

**Pull request merged**